src/app/qgslabelinggui.cpp

```diff
@@ -281,7 +263,51 @@ QgsLabelingGui::QgsLabelingGui( QgsVectorLayer* layer, QgsMapCanvas* mapCanvas,
 
   mLabelingOptionsListWidget->setCurrentRow( settings.value( QString( "/Windows/Labeling/Tab" ), 0 ).toInt() );
   groupBox_mPreview->setVisible( false );
+
   setDockMode( false );
+
+  Q_FOREACH ( QgsDataDefinedButton* button, findChildren<QgsDataDefinedButton*>() )
+  {
+    connect( button, SIGNAL( dataDefinedActivated( bool ) ), this, SLOT( updatePreview() ) );
+    connect( button, SIGNAL( dataDefinedChanged( QString ) ), this, SLOT( updatePreview() ) );
+  }
+
+  Q_FOREACH ( QgsFieldExpressionWidget* widget, findChildren<QgsFieldExpressionWidget*>() )
+  {
+    connect( widget, SIGNAL( fieldChanged( QString ) ), this, SLOT( updatePreview() ) );
+  }
+
+  Q_FOREACH ( QComboBox* widget, findChildren<QComboBox*>() )
+  {
+    connect( widget, SIGNAL( currentIndexChanged( int ) ), this, SLOT( updatePreview() ) );
+  }
+
+  Q_FOREACH ( QSpinBox* widget, findChildren<QSpinBox*>() )
+  {
+    connect( widget, SIGNAL( valueChanged( int ) ), this, SLOT( updatePreview() ) );
+  }
+
+  Q_FOREACH ( QDoubleSpinBox* widget, findChildren<QDoubleSpinBox*>() )
+  {
+    connect( widget, SIGNAL( valueChanged( double ) ), this, SLOT( updatePreview() ) );
+  }
+
+  Q_FOREACH ( QgsColorButtonV2* widget, findChildren<QgsColorButtonV2*>() )
+  {
+    connect( widget, SIGNAL( colorChanged( QColor ) ), this, SLOT( updatePreview() ) );
+  }
+
+  Q_FOREACH ( QCheckBox* widget, findChildren<QCheckBox*>() )
+  {
+    connect( widget, SIGNAL( toggled( bool ) ), this, SLOT( updatePreview() ) );
+  }
+
+  Q_FOREACH ( QRadioButton* widget, findChildren<QRadioButton*>() )
+  {
+    connect( widget, SIGNAL( toggled( bool ) ), this, SLOT( updatePreview() ) );
+  }
+
+  connect( mQuadrantBtnGrp, SIGNAL( buttonClicked( int ) ), this, SLOT( updatePreview() ) );
 }
```

Comment:
@nyalldawson @wonder-sk @m-kuhn keen for your thoughts on this.  Seems to work well and is a quick way to connect all the widgets to the auto apply function without having to list them all.  Adds a bit of future proofing as well so you don't have to maintain the code for that part.